.jenkins/jobs/webapp_deployment

```diff
@@ -4,46 +4,106 @@ pipeline {
     parameters {
         string(name: 'use_case_base_path', defaultValue: 'named_entity_recognition', description: 'The flow use-case to execute')
         string(name: 'deploy_environment', defaultValue: 'dev', description: 'Execution Environment')
+        string(name: 'env_vars', description: 'The API key for authentication')
     }
 
     environment {
-        dev_connections = credentials('COMMON_DEV_CONNECTIONS')
         registry_details = credentials('DOCKER_IMAGE_REGISTRY')
     }
 
     stages {
-    
-    // Create local promptflow connections
-    stage('Create local promptflow connections') {
+
+    stage('Setup Environment') {
+
             steps {
-                installRequirements('execute_job_requirements')
-                installRequirements('build_validation_requirements')
-                withCredentials([azureServicePrincipal('AZURE_CREDENTIALS')]) {
-                    withPythonEnv('/usr/bin/python3.9') {
-                    sh '''
-                    python -m llmops.common.prompt_local_connections \\
-                        --base_path ${use_case_base_path} \\
-                        --env_name ${deploy_environment} \\
-                        --connection_details "$dev_connections"
-                    '''
+                script {
+                    // Read env_vars content
+                    def envFileContent = params.env_vars.trim()
+
+                    // Split content into lines
+                    def lines = envFileContent.readLines()
+
+                    // Iterate through each line
+                    lines.each { line ->
+                        // Split line by '=' to get key and value
+                        def (key, value) = line.split('=')
+
+
+                        // Set environment variable
+                        env."$key" = value
+                    }
                 }
             }
+        }
+
+    // Create .env file again for verification
+        stage('Load .env file') {
+            steps {
+                script {
+                    // Ensure the workspace directory is writable
+                    sh 'chmod -R 777 $WORKSPACE'
+
+                    // Use the secret file credentials
+                    withCredentials([file(credentialsId: 'ENV_VARS', variable: 'ENV_FILE')]) {
+                        // Check if the secret file exists and read its content
+                        sh 'ls -l $ENV_FILE'
+                        def envContent = readFile(file: ENV_FILE)
+
+                        // Write the content to .env file in the workspace
+                        writeFile file: '.env', text: envContent
+
+                        // Check if the .env file was created
+                        sh 'ls -l .env''
+
+                        // Optionally, load the .env file using Python
+                        def loadEnvScript = """
+import os
+from dotenv import load_dotenv
+
+# Load the .env file
+load_dotenv('.env')
```

Comment:
Nit: could this code block be at the same level as the rest of the script?

Reply:
@machteldbogels  the issue is the python code block , doesn't work if indented further. It throws an error due to indention issue. I checked with Jenkins pipeline linter and it's expected to be this way.
But, I will reverify this again and share a screenshot

Reply:
<img width="1472" alt="Screenshot 2024-06-17 at 18 21 05" src="https://github.com/microsoft/llmops-promptflow-template/assets/19162354/7d250927-f886-4dbe-9258-eb94ea6791bc">
Screenshot attached shows jenkins pipeline validation with the indentation in the PR

Reply:
If I change the indent to below , see error
<img width="813" alt="Screenshot 2024-06-17 at 19 16 06" src="https://github.com/microsoft/llmops-promptflow-template/assets/19162354/bc925083-935b-4749-9d56-88ef973d74f9">
<img width="886" alt="Screenshot 2024-06-17 at 19 15 56" src="https://github.com/microsoft/llmops-promptflow-template/assets/19162354/886889c9-db1d-4eb0-8f47-6995382565dd">